lib/steep/interface/builder.rb

```diff
@@ -725,6 +727,37 @@ def proc_shape(proc, proc_shape)
         shape
       end
 
+      def replace_guard_method(definition, method_def, method_type)
+        match = method_def.annotations.filter_map { AST::Types::Logic::Guard::PATTERN.match(_1.string) }.first
+        if match
+          subject = match[1] or raise
+          operator = match[2] or raise
+          type_name = match[3] or raise
+
+          type = RBS::Parser.parse_type(type_name)
+          raise "unknown type: #{type_name}" unless type
+
+          context = context_from(definition.type_name)
+          type = type.map_type_name { factory.absolute_type_name(_1, context: context) or raise }
+          guard = AST::Types::Logic::Guard.new(subject: subject, operator: operator, type: type)
+          definition.type_name
+          method_type.with(
+            type: method_type.type.with(return_type: guard)
+          )
+        else
+          method_type
+        end
+      end
+
+      def context_from(type_name)
+        if type_name.namespace == RBS::Namespace.root
+          [nil, type_name]
+        else
+          parent = context_from(type_name.namespace.to_type_name)
+          [parent, type_name]
+        end
+      end
```

Comment:
I could not find such a helper method in Steep and RBS.
I'm not sure where the best place is.